addons/payment/static/src/js/payment_form.js

```diff
@@ -0,0 +1,256 @@
+                var form_data = this.getFormData($('input', acquirer_form));
+                var ds = $('input[name="data_set"]', acquirer_form)[0];
+
+                // we force the check when adding a card trough here
+                form_data.verify_validity = true;
+
+                // do the call to the route stored in the 'data_set' input of the acquirer form, the data must be called 'create-route'
+                ajax.jsonRpc(ds.dataset.createRoute, 'call', form_data).then(function(data) {
+                    // if the server has returned true
+                    if(data.result) {
+                        // and it need a 3DS authentication
+                        if(data['3d_secure'] !== false) {
+                            // then we display the 3DS page to the user
+                            $("body").html(data['3d_secure']);
+                        }
+                        // if it doesn't require 3DS
+                        else {
+                            // we just go to the return_url or reload the page
+                            if(form_data.return_url) {
+                                window.location = form_data.return_url;
+                            }
+                            else {
+                                window.location.reload();
+                            }
+                        }
+                    }
+                    // if the server has returned false, we display an error
+                    else {
+                        self.error(core._t('Error'),
+                            core._t("<p>We are not able to add your payment method at the moment.</p>"));
+                    }
+                    // here we remove the 'processing' icon from the 'add a new payment' button
+                    $(button).attr('disabled', false);
+                    $(button).find('span.o_loader').remove();
+                }).fail(function(message, data) {
+                    // if the rpc fails, pretty obvious
+                    $(button).attr('disabled', false);
+                    $(button).find('span.o_loader').remove();
+
+                    self.error(core._t('Error'),
+                        core._t("<p>We are not able to add your payment method at the moment.</p>") + (core.debug ? data.data.message : ''));
+                });
+            }
+            else {
+                this.error(core._t('Error'),
+                    core._t('<p>Please select the option to add a new payment method</p>'));
+            }
+        },
+        // event handler when clicking on a button to delete a payment method
+        deletePmEvent: function (ev) {
+            ev.stopPropagation();
+            ev.preventDefault();
+            var self = this;
+            var pm_id = ev.target.value;
+
+            var Delete = function() {
+                ajax.jsonRpc('/payment/delete/', 'call', {'pm_id':pm_id}).then(function(data) {
```

Comment:
Something like this?

        rpc.query({
                model: 'payment.token',
                method: 'unlink',
                args: [[parseInt(pm_id)]],
        })